include/deal.II/lac/la_vector.templates.h

```diff
@@ -398,7 +398,7 @@ namespace LinearAlgebra
 
     // in >> c;
     in.read(&c,1);
-    AssertThrow(c=']', ExcIO());
+    AssertThrow(c==']', ExcIO());
```

Comment:
Ouch! Do we have the same bug in the regular `Vector` and `BlockVector` classes?